File: src/IsbInterop/FatalIsbInteropException.vb
''' <summary>
''' Фатальное исключение.
''' </summary>
Public Class FatalIsbInteropException
    Inherits Exception
    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="message">Сообщение.</param>
    Public Sub New(message As String)
        MyBase.New(message)
    End Sub

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="message">Сообщение.</param>
    ''' <param name="innerException">Внутреннее исключение.</param>
    Public Sub New(message As String, innerException As Exception)
        MyBase.New(message, innerException)
    End Sub
End Class

File: src/IsbInterop/FatalIsbInteropException.cs
using System;

namespace IsbInterop
{
  /// <summary>
  /// Фатальное исключение.
  /// </summary>
  public class FatalIsbInteropException : Exception
  {
    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="message">Сообщение.</param>
    public FatalIsbInteropException(string message)
      : base(message) { }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="message">Сообщение.</param>
    /// <param name="innerException">Внутреннее исключение.</param>
    public FatalIsbInteropException(string message, Exception innerException)
      : base(message, innerException) { }
  }
}
